thrift/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -689,9 +693,10 @@ private static HttpData encodeSuccess(ServiceRequestContext ctx,
                                           SerializationFormat serializationFormat,
                                           String methodName, int seqId,
                                           TBase<?, ?> result) {
-
-        final TMemoryBuffer buf = new TMemoryBuffer(128);
-        final TProtocol outProto = ThriftProtocolFactories.get(serializationFormat).getProtocol(buf);
+        final ByteBufAllocator alloc = ctx.alloc();
+        ByteBuf buf = alloc.buffer(128);
```

Comment:
That's a good question.

I thought `buffer()` always allocates a heap buffer, but it seems like I was wrong. (and I think I wrote this!) Looking from the code, the difference between `buffer()` and `ioBuffer()` is that `buffer()` respects the `io.netty.noPreferDirect` system property and `ioBuffer()` does not. They both falls back to heap buffers when `sun.misc.Unsafe` is unavailable though.

So.. I think `ioBuffer()` could be just deprecated? Let me discuss about this with Netty folks.